### PR TITLE
Batching using pytorch Dataset and DataLoader

### PR DESCRIPTION
**Pull request by ynashed**
Currently functioning but still needs some work:

- [ ] `optimize/sim_module.py` needs to be finished as a pytorch module
- [ ] testing

### CODE REVIEW

File: optimize/example_run.py [hunk 1]
```diff
@@ -1,27 +1,63 @@
...
+import traceback
+from torch.utils.data import DataLoader
 
-def main(param_list, input_file, lr, epochs, seed):
-    param_fit = ParamFitter(param_list, lr=lr) 
-    param_fit.make_target_sim(seed=seed)
-    param_fit.load_data(filename=input_file)
-    param_fit.fit(epochs=epochs)
+from .fit_params import ParamFitter
+from .dataio import TracksDataset
```
Comment (sgasioro) [inline optimize/example_run.py:9]: I got a `ImportError: attempted relative import with no known parent package` here -- maybe we should make an `optimize/optimize` directory and move everything (except `example_run`) there?

Comment (ynashed) [reply optimize/example_run.py:9]: You should be able to run it as a python module using `-m` from the parent directory. Something like
```
python -u -m optimize.example_run --params lifetime --input_file data/edepsim-output.h5 --batch_sz 2
```

**Comment by sgasioro** [comment]
Other than the small comment, looks good! As discussed last week, a couple of things we may want to implement still for the data loading, but @YifanC can add in some of her structure

File: optimize/example_run.py (hunk 1 above)
Comment (sgasioro) [reply optimize/example_run.py:9]: Ah ok ok, that's fine then

**Pull request merged**